app/client/cli/debug.go

```diff
@@ -44,15 +46,17 @@ var (
 		PromptSendBlockRequest,
 	}
 
-	// validators holds the list of the validators at genesis time so that we can use it to create a debug address book provider.
-	// Its purpose is to allow the CLI to "discover" the nodes in the network. Since currently we don't have churn and we run nodes only in LocalNet, we can rely on the genesis state.
-	// HACK(#416): This is a temporary solution that guarantees backward compatibility while we implement peer discovery
-	validators []*coreTypes.Actor
-
 	configPath  string = getEnv("CONFIG_PATH", "build/config/config1.json")
 	genesisPath string = getEnv("GENESIS_PATH", "build/config/genesis.json")
 )
 
+type ctxKey int
```

Comment:
I made a mistake here, I should have injected the `bus` into the cobra context and not the individual modules.
That's a pretty common pattern that leverages the Context https://pkg.go.dev/context to pass around stuff that is required within the context of a request (HTTP or like in the case of a CLI, a user interaction) something that otherwise we have to explicitly add as an argument along the whole chain of function calls.

It's a pretty common pattern.

The problem is visible here: https://github.com/pokt-network/pocket/compare/issue/416-kiss-peer-discovery-k8s-base354...issue/416-cli-ctx-2#diff-d788eb33d67196047ff542eb76cc2be8f0d00cc1c3ee44bc77aa3192b1071456R96-R102

and this is probably a better version of the solution using the CLI context: https://github.com/pokt-network/pocket/compare/issue/416-kiss-peer-discovery-k8s-base354...issue/416-cli-ctx-1

Reply:
Resolving in https://github.com/pokt-network/pocket/pull/521/files#r1109158407